test/test_pipe_baseline.py

```diff
@@ -1,21 +1,73 @@
 import unittest
+import resonance.pipe as pipe
+import resonance.si as si
+import resonance.db as db
 import resonance.run as run
+import numpy as np
 
 
 class TestPipeBaseline(unittest.TestCase):
-    def check(self, si, blocks, expected):
-        code = '''
-from resonance import *
-in = input(1)
-out = pipe.baseline(in)
-createOutput(out, 'out')
-'''
+    def check(self, si, blocks, expected, beginOffset, endOffset):
+
+        code = "from resonance import *"\
+               "inputData = input(0)"\
+               "out = pipe.baseline(inputData, {0}, {1})"\
+               "createOutput(out, 'out')"\
+               .format(beginOffset, endOffset)
 
         online = run.online([si], blocks, code)
         offline = run.offline([si], blocks, code)
 
         self.assertEqual(online, expected)
         self.assertEqual(online, offline)
 
-    def test_spatial(self):
-        pass
+    def test_baseline(self):
+        channels = 2
+        samples = 20
+        time = 3
+
+        w_si = si.Window(channels, samples, 250)
+        data = np.arange(1, 41).reshape((samples, channels))
+
+        srcWindow = db.Window(w_si, time, data)
+
+        expectedData = np.array([-8.5, -7.5, -6.5, -5.5, -4.5, -3.5, -2.5, -1.5, -0.5, 0.5, 1.5, 2.5, 3.5, 4.5, 5.5,
```

Comment:
This could be written as
`data - np.asarray([-9.5, -9.5])`
that could help to keep the code size in the test smaller

also here for both channels average value is the same, can you make it different?

Reply:
The current shape of this array is incorrect (1 channel)